### PR TITLE
virtual/rust: Add 1.77.1

### PR DESCRIPTION
**Pull request by bowlofeggs**

### CODE REVIEW

File: sys-devel/rust-std/rust-std-1.77.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+RDEPEND="${DEPEND}"
+
+# need full compiler to run tests
+RESTRICT="test"
+
+QA_FLAGS_IGNORED="usr/lib/rust/${PV}/rustlib/.*/lib/lib.*.so"
+
+PATCHES=(
+	"${FILESDIR}"/1.75.0-handle-vendored-sources.patch  # remove for >=1.77.0
+)
```
Comment (Jamim) [inline sys-devel/rust-std/rust-std-1.77.0.ebuild:41]: Hello @bowlofeggs,
It looks like this patch isn't needed anymore.
```suggestion
```

**Review by Jamim — CHANGES_REQUESTED** [review]

File: sys-devel/rust-std/rust-std-1.77.0.ebuild (hunk 1 above)
Comment (bowlofeggs) [reply sys-devel/rust-std/rust-std-1.77.0.ebuild]: Thanks! Fixed in https://github.com/gentoo/gentoo/pull/35866/commits/87927be1014b8c8c24e1c40c7d8ba23c27086e7d

**Review by Jamim — APPROVED** [review]

**Comment by Jamim** [comment]
_Hello @thesamesam,_

It would be nice to have Rust 1.77.0 in the Portage tree.
Would you mind reviewing this PR?

_Thanks in advance!_

**Comment by thesamesam** [comment]
I'm happy to review and merge things if they look stuck, but please don't ping me on everything and immediately unless it looks like it needs some intervention.

**Comment by bowlofeggs** [comment]
There was a 1.77.1 release this week which sounds like it is entirely Windows oriented, but I figured I'd go ahead and bump to it since we haven't gotten too far with testing on this PR yet.

**Comment by mgorny** [comment]
FYI, this is now blocking `dev-python/uv`, sigh. I hate software engineering these days.

**Comment by thesamesam** [comment]
I'll build my usual Rust packages with it and if it works we can probably ship it, I guess.

**Comment by thesamesam** [comment]
> I'll build my usual Rust packages with it and if it works we can probably ship it, I guess.

All built OK. Any objections with merging later today?

**Comment by bowlofeggs** [comment]
> All built OK. Any objections with merging later today?

None from me. I didn't quite get through the entire test matrix, but I did get about 40%ish of the way through and things seem fine.

**Comment by bowlofeggs** [comment]
I have completed the testing, though on `x86_64` only. I don't have hardware to test other architectures.

**Comment by matoro** [comment]
Looks good on alt-arches.  Tested the following:
* `USE=-system-bootstrap` build
* `sys-apps/ripgrep` testsuite
* `USE=system-bootstrap` build

Tested:  arm64, arm, ppc64le, ppc64, ppc, sparc (except `USE=-system-bootstrap`), riscv (except `USE=system-bootstrap`), loong

**Comment by bowlofeggs** [comment]
> @bowlofeggs some nitpicking: I believe IUSE should always be in alphabetical order, you swapped it around in virtual/rust-1.77.1

I didn't intentionally swap it. I think what happened is that I started from 1.76.0 from [this commit](https://github.com/gentoo/gentoo/commit/9e8e96ef78ad1e9dfbd84587c45e80e4922d80f5), which had it like this:

https://github.com/gentoo/gentoo/blob/9e8e96ef78ad1e9dfbd84587c45e80e4922d80f5/virtual/rust/rust-1.76.0.ebuild#L17

Then it looks like Sam later made this change, which flipped the order of those two parameters:

https://github.com/gentoo/gentoo/commit/ec95eaffca8731e4604cacefe2fabc4b814f7a9c

I believe that change was made after I had already copied the previous commit's file, and I didn't notice the change after that.

I guess this is one of the consequences of each version bump being a new file (i.e., the git diff doesn't show what changed between 1.76 and 1.77 since 1.77 is a new file.)

All that said, feel free to send a PR to flip it back if you like. I do find alphabetical order pleasing.

**Comment by bowlofeggs** [comment]
One other thing I'd like to see here if I can make the time to get it done is to add -9999 versions for these files. Then we can use them for "development" changes, and then changes like this *would* show up in git diffs. I think if we'd had that then I wouldn't have re-reversed those two ☺

**Pull request closed without merging**